cpp/benchmarks/stream_compaction/distinct.cpp

```diff
@@ -40,27 +69,29 @@ void nvbench_distinct(nvbench::state& state, nvbench::type_list<Type>)
 
   state.set_cuda_stream(nvbench::make_cuda_stream_view(cudf::get_default_stream().value()));
   state.exec(nvbench::exec_tag::sync, [&](nvbench::launch& launch) {
-    auto result = cudf::distinct(input_table,
-                                 {0},
-                                 cudf::duplicate_keep_option::KEEP_ANY,
-                                 cudf::null_equality::EQUAL,
-                                 cudf::nan_equality::ALL_EQUAL);
+    auto result = cudf::distinct(
+      input_table, {0}, keep, cudf::null_equality::EQUAL, cudf::nan_equality::ALL_EQUAL);
   });
 }
 
-using data_type = nvbench::type_list<bool, int8_t, int32_t, int64_t, float, cudf::timestamp_ms>;
+using data_type = nvbench::type_list<int32_t, int64_t>;
 
 NVBENCH_BENCH_TYPES(nvbench_distinct, NVBENCH_TYPE_AXES(data_type))
   .set_name("distinct")
   .set_type_axes_names({"Type"})
-  .add_int64_axis("NumRows", {10'000, 100'000, 1'000'000, 10'000'000});
+  .add_string_axis("keep", {"any", "first", "last", "none"})
+  .add_int64_axis("cardinality",
+                  {100, 1'000, 10'000, 100'000, 1'000'000, 10'000'000, 100'000'000, 1'000'000'000})
+  .add_int64_axis("NumRows",
+                  {100, 1'000, 10'000, 100'000, 1'000'000, 10'000'000, 100'000'000, 1'000'000'000});
```

Comment:
we adding many more benchmark cases by doing so. maybe worth considering fewer cases for a shorter default execution time
```suggestion
  .add_string_axis("keep", {"any", "first", "last", "none"})
  .add_int64_axis("cardinality",
                  {100, 100'000, 10'000'000, 1'000'000'000})
  .add_int64_axis("NumRows",
                  {100, 100'000, 10'000'000, 1'000'000'000});
```
not a hard requirement though

Reply:
@bdice I support @PointKernel 's suggestion. `distinct` just took me >15 minutes to run due to the wider axes.